### PR TITLE
Pre-processing: Resize Linear C1 refactoring.

### PR DESCRIPTION
**Pull request by anna-khakimova**
### Details:
 - *Resize 8UC1 refactoring*
 - *Resize 32FC1 refactoring*

### CODE REVIEW

File: inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp [hunk 1]
```diff
@@ -939,6 +500,222 @@ struct typed_i420_to_rgb_row {
...
+                    outW * sizeof(index_t)     +
+                    outH * sizeof(alpha_t)     +
+                    outH * sizeof(index_t) * 2 +
+                     inW * sizeof(T) * lpi * chanNum;
+
+        return static_cast<int>(size);
+    }
+};
+static inline double invRatio(int inSz, int outSz) {
+    return static_cast<double>(outSz) / inSz;
```
Comment (sivanov-work) [inline inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp:561]: Can `inSz` be equal 0 ? may be we need to put assert here

File: inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp [hunk 2]
```diff
@@ -939,6 +500,222 @@ struct typed_i420_to_rgb_row {
...
+}
+
+static inline double ratio(int inSz, int outSz) {
+    return 1 / invRatio(inSz, outSz);
```
Comment (sivanov-work) [inline inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp:565]: and here

File: inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp [hunk 3]
```diff
@@ -939,6 +507,313 @@ struct typed_i420_to_rgb_row {
...
+
+    template<typename tag = isa_tag_t>
+    typename std::enable_if<!std::is_same<tag, scalar_tag>::value, p_f>::type
+        operator()(type_to_type<float>) {
+        return [](float* dst[], const float* src0[], const float* src1[],
+                  const float alpha[], const int mapsx[],
+                  const float beta[], const Size& inSz,
+                  const Size& outSz, const int lpi, const int length) {
+                calcRowLinear32FC1Impl(isa_tag_t{}, dst, src0, src1, alpha,
+                                      mapsx, beta, inSz, outSz, lpi, length);
```
Comment (mpashchenkov) [inline inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp:799]: One more space.

File: inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp [hunk 4]
```diff
@@ -1005,54 +880,292 @@ GAPI_FLUID_KERNEL(FI420toRGB, I420toRGB, false) {
...
 };
-};
 
-namespace {
-struct ColorConversionISA {
+GAPI_FLUID_KERNEL(FSplit2, Split2, false) {
+    static const int LPI = 4;
+    static const int Window = 1;
+    static void run(const cv::gapi::fluid::View & in,
+        cv::gapi::fluid::Buffer & out1,
```
Comment (mpashchenkov) [inline inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp:888]: Alignment.

Comment (mpashchenkov) [reply inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp:888]: And the same comment about other run methods.

File: inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp [hunk 5]
```diff
@@ -1005,54 +880,292 @@ GAPI_FLUID_KERNEL(FI420toRGB, I420toRGB, false) {
...
+        GAPI_DbgAssert(in.meta().depth == out1.meta().depth);
+        GAPI_DbgAssert(in.meta().depth == out2.meta().depth);
+        GAPI_DbgAssert(in.meta().depth == out3.meta().depth);
+
+        GAPI_DbgAssert(is_cv_type_in_list<split_supported_types>(in.meta().depth));
+
+        const auto rowFunc = type_dispatch<split_supported_types>(in.meta().depth, cv_type_id{}, typed_split_row<isa_tag_t, 3>{}, nullptr);
+        for (int i = 0, lpi = out1.lpi(); i < lpi; i++) {
+            std::array<uint8_t*, 3> outs = { out1.OutLineB(i), out2.OutLineB(i),
+                                            out3.OutLineB(i) };
```
Comment (mpashchenkov) [inline inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp:925]: One more space.

File: inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp [hunk 6]
```diff
@@ -1005,54 +880,292 @@ GAPI_FLUID_KERNEL(FI420toRGB, I420toRGB, false) {
...
+        GAPI_DbgAssert(in.meta().depth == out1.meta().depth);
+        GAPI_DbgAssert(in.meta().depth == out2.meta().depth);
+        GAPI_DbgAssert(in.meta().depth == out3.meta().depth);
+        GAPI_DbgAssert(in.meta().depth == out4.meta().depth);
+        GAPI_DbgAssert(is_cv_type_in_list<split_supported_types>(in.meta().depth));
+
+        const auto rowFunc = type_dispatch<split_supported_types>(in.meta().depth, cv_type_id{}, typed_split_row<isa_tag_t, 4>{}, nullptr);
+        for (int i = 0, lpi = out1.lpi(); i < lpi; i++) {
+            std::array<uint8_t*, 4> outs = { out1.OutLineB(i), out2.OutLineB(i),
+                                            out3.OutLineB(i), out4.OutLineB(i) };
```
Comment (mpashchenkov) [inline inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp:953]: And here

**Review by mpashchenkov — COMMENTED** [review]
Alignment.

**Comment by anna-khakimova** [comment]
PR https://github.com/openvinotoolkit/openvino/pull/6330 contains this patch. It was closed because of a large number of merge conflicts.

**Pull request closed without merging**